tests/components/statistics/test_sensor.py

```diff
@@ -1561,18 +1561,12 @@ async def test_initialize_from_database_with_maxage(
     """Test initializing the statistics from the database."""
     current_time = dt_util.utcnow()
 
-    # Testing correct retrieval from recorder, thus we do not
-    # want purging to occur within the class itself.
-    def mock_purge(self, *args):
-        return
-
     # enable and pre-fill the recorder
     await hass.async_block_till_done()
     await async_wait_recording_done(hass)
 
     with (
         freeze_time(current_time) as freezer,
-        patch.object(StatisticsSensor, "_purge_old_states", mock_purge),
```

Comment:
Why is this changed?

Reply:
Moving the "purge" functionality into the separate buffer class makes it no longer available at StatisticsSensor level. Preventing the purge from happening wasn't necessary anyway, since no value would be purged in the test as it is now. It could have been necessary in the past in case the test was changed.